pygments/lexers/matlab.py

```diff
@@ -142,9 +142,10 @@ class MatlabLexer(RegexLexer):
             (r"[^']*'", String, '#pop'),
         ],
         'commandargs': [
+            (r"\s+", Text),
             ("'[^']*'", String),
             ("[^';\n]+", String),
-            (";?\n?", Punctuation, '#pop'),
+            (";?(?=\n)?", Punctuation, '#pop'),
```

Comment:
I think this doesn't do anything, since the lookahead is optional.

I guess use `bygroups()` instead to have the `\n` as Text.

Reply:
With the lookahead the \n will never be labeled as Punctuation, but a single \n will correctly #pop the state (without being consumed).

Reply:
But since the lookahead is optional, I don't see what the difference is vs. removing it completely.

Reply:
Because otherwise you stay in commandargs mode even after the newline.

Reply:
Again, I claim that you can remove `(?=\n)?` and there is no observable difference. I may be missing the forest for the trees, so do you have an example that shows otherwise?

Reply:
A bad case is
```
In [1]: print(list(pygments.lex('foo bar baz\nquux', MatlabLexer())))
[(Token.Name, 'foo'), (Token.Text, ' '), (Token.Literal.String, 'bar baz'), (Token.Text, '\n'), (Token.Literal.String, 'quux'), (Token.Text, '\n'), (Token.Punctuation, '')]
```
note how the `quux` on the second line is also interpreted as a Literal.String rather than a Name.
...
except that I realized that even with the (?=\n)? this still fails because the \s+ coming first globs everything; and there's another bug (`bar baz` is two strings).

I think
```
        'commandargs': [
            (r"[ \t]+", Text),
            ("'[^']*'", String),
            (r"[^';\s]+", String),
            (";?", Punctuation, '#pop'),
        ]
```
may work better?

Reply:
Yes, that should leave commandargs mode after anything that isn't space, string or word.

Reply:
pushed